dirsrvtests/tests/suites/replication/regression_test.py

```diff
@@ -924,6 +948,109 @@ def test_moving_entry_make_online_init_fail(topology_m2):
+        for ret in entries:
+            log.debug("Found keepalive entry:\n"+str(ret));
+    return entries
+
+def verify_keepalive_entries(topo, expected):
+    #Check that keep alive entries exists (or not exists) for every masters on every masters
+    #Note: The testing method is quite basic: counting that there is one keepalive entry per master.
+    # that is ok for simple test cases like test_online_init_should_create_keepalive_entries but
+    # not for the general case as keep alive associated with no more existing master may exists
+    # (for example after: db2ldif / demote a master / ldif2db / init other masters)
+    # ==> if the function is somehow pushed in lib389, a check better than simply counting the entries
+    # should be done.
+    for masterId in topo.ms:
+        master=topo.ms[masterId]
+        for replica in Replicas(master).list():
+            if (replica.get_role() != ReplicaRole.MASTER):
+               continue
+            replica_info = f'master: {masterId} RID: {replica.get_rid()} suffix: {replica.get_suffix()}'
+            log.debug(f'Checking keepAliveEntries on {replica_info}')
+            keepaliveEntries = get_keepalive_entries(master, replica);
+            expectedCount = len(topo.ms) if expected else 0
+            foundCount = len(keepaliveEntries)
+            if (foundCount == expectedCount):
+                log.debug(f'Found {foundCount} keepalive entries as expected on {replica_info}.')
+            else:
+                log.error(f'{foundCount} Keepalive entries are found '
+                          f'while {expectedCount} were expected on {replica_info}.')
+                assert False
+
+
+def test_online_init_should_create_keepalive_entries(topo_m2):
+    """Check that keep alive entries are created when initializinf a master from another one
+
+    :id: d5940e71-d18a-4b71-aaf7-b9185361fffe
+    :setup: Two masters replication setup
+    :steps:
+        1. Generate ldif without replication data
+        2  Init both masters from that ldif
+        3  Check that keep alive entries does not exists
+        4  Perform on line init of master2 from master1
+        5  Check that keep alive entries exists
+    :expectedresults:
+        1. No error while generating ldif
+        2. No error while importing the ldif file
+        3. No keepalive entrie should exists on any masters
+        4. No error while initializing master2
+        5. All keepalive entries should exist on every masters
+
+    """
+
+    m1 = topo_m2.ms["master1"]
+    m2 = topo_m2.ms["master2"]
+    # Step 1: Generate ldif without replication data
+    m1.stop()
+    m2.stop()
+    ldif_file = '%s/norepl.ldif' % m1.get_ldif_dir()
+    m1.db2ldif(bename=DEFAULT_BENAME, suffixes=[DEFAULT_SUFFIX],
+               excludeSuffixes=None, repl_data=False,
+               outputfile=ldif_file, encrypt=False)
+    # Remove replication metadata that are still in the ldif
+    _remove_replication_data(ldif_file)
```

Comment:
One out of three runs has failed here because it didn't find the `ldif_file`.
Maybe, it's worth to set a small `time.wait()` after `db2ldif` task.

Reply:
Strange:   As far as I remember the off line export is supposed to be synchronous.
  (That is not an ldap task that need monitoring) but it is a ns-slapd instance in db2ldif mode)  
So I am not sure that adding a delay will help.
I am running the test in loop to try to reproduce the failure

Reply:
So far I did not yet reproduced the issue but I hit another one: step 5 is sometime failing because the keep alive entry was not yet replicated +=> Must wait that master1 is in sync with master2 before verifying the entries.
Restarted the tests after adding a wait_for_ruv(m2,m1)

Reply:
Test run in loop without error since 2 hours. I also double checked and indeed off-line export is synchronous.
So I do not know what happened in your environment (At this point: either the file was not created or it has been removed, both scenario let me think that something unusual happened in your environment (maybe the instance was been cleared by another test or some cleanup task in background ... ))

Reply:
> Test run in loop without error since 2 hours. I also double checked and indeed off-line export is synchronous.
> So I do not know what happened in your environment (At this point: either the file was not created or it has been removed, both scenario let me think that something unusual happened in your environment (maybe the instance was been cleared by another test or some cleanup task in background ... ))

Looks like this.
I test on an isolated environment (freshly created libvirt machine) but still, something could happen there.

If the test will fail on our CI, we will see. The failure was purely test-related (and, probably, env-related) so no worries :)

You have my ack! And I think we can merge.